keras_nlp/tokenizers/unicode_character_tokenizer.py

```diff
@@ -0,0 +1,276 @@
+        normalization_form: One of the following string values (None, 'NFC',
+            'NFKC', 'NFD', 'NFKD'). If set will normalize unicode to the given
+            form before tokenizing.
+        errors: One of ('replace', 'remove', 'strict'). Specifies the
+            `detokenize()` behavior when an invalid codepoint is encountered.
+            (same behavior as
+            https://www.tensorflow.org/api_docs/python/tf/strings/unicode_transcode)
+        replacement_char: The unicode codepoint to use in place of invalid
+            codepoints. Defaults to 65533 (U+FFFD).
+        input_encoding: The encoding of the input text. Defaults to "utf-8".
+        output_encoding: The encoding of the output text. Defaults to "utf-8".
+
+    Examples:
+
+    Basic Usage.
+    >>> inputs = "Unicode Tokenizer"
+    >>> tokenizer = keras_nlp.tokenizers.UnicodeCharacterTokenizer()
+    >>> tokenizer(inputs)
+    <tf.Tensor: shape=(17,), dtype=int32, numpy=
+    array([117, 110, 105,  99, 111, 100, 101,  32, 116, 111, 107, 101, 110,
+        105, 122, 101, 114], dtype=int32)>
+
+    Ragged outputs.
+    >>> inputs = ["Ninja", "Samurai", "पुस्तक", "کتاب"]
+    >>> tokenizer = keras_nlp.tokenizers.UnicodeCharacterTokenizer()
+    >>> tokenizer(inputs)
+    <tf.RaggedTensor [[110, 105, 110, 106, 97],
+        [115, 97, 109, 117, 114, 97, 105],
+        [2346, 2369, 2360, 2381, 2340, 2325], 
+        [1705, 1578, 1575, 1576]]>
+
+    Dense outputs.
+    >>> inputs = ["Ninja", "Samurai", "पुस्तक", "کتاب"]
+    >>> tokenizer = keras_nlp.tokenizers.UnicodeCharacterTokenizer(
+        sequence_length=8)
+    >>> tokenizer(inputs)
+    <tf.Tensor: shape=(2, 8), dtype=int32, numpy=
+    array([[ 110,  105,  110,  106,   97,    0,    0,    0],
+       [ 115,   97,  109,  117,  114,   97,  105,    0],
+       [2346, 2369, 2360, 2381, 2340, 2325,    0,    0],
+       [1705, 1578, 1575, 1576,    0,    0,    0,    0]], dtype=int32)>
+
+    Tokenize first, then batch the dataset up.
+    >>> inputs = ["Ninja", "Samurai", "पुस्तक", "کتاب"]
+    >>> tokenizer = keras_nlp.tokenizers.UnicodeCharacterTokenizer()
+    >>> ds = tf.data.Dataset.from_tensor_slices(inputs)
+    >>> ds = ds.map(tokenizer)
+    >>> ds = ds.apply(tf.data.experimental.dense_to_ragged_batch(3))
+    >>> ds.take(1).get_single_element()
+    <tf.RaggedTensor [[110, 105, 110, 106, 97], 
+        [115, 97, 109, 117, 114, 97, 105],
+        [2346, 2369, 2360, 2381, 2340, 2325]]>
+
+    Batch up the inputs and then tokenize.
+    >>> inputs = ["Ninja", "Samurai", "पुस्तक", "کتاب"]
+    >>> tokenizer = keras_nlp.tokenizers.UnicodeCharacterTokenizer()
+    >>> ds = tf.data.Dataset.from_tensor_slices(inputs)
+    >>> ds = ds.batch(3).map(tokenizer)
+    >>> ds.take(1).get_single_element()
+    <tf.RaggedTensor [[110, 105, 110, 106, 97], 
+        [115, 97, 109, 117, 114, 97, 105],
+        [2346, 2369, 2360, 2381, 2340, 2325]]>
+
+    Tokenize first, then batch the dataset up for Dense Outputs
```

Comment:
same, "batch the dataset for dense outputs". remove "up"

A couple more "up"s below as well